wasapi_client.py

```diff
@@ -0,0 +1,366 @@
+                           help='number of WARC downloading processes')
+    out_group.add_argument('-s',
+                           '--size',
+                           action='store_true',
+                           help='print count and total size of files and exit')
+
+    # Arguments to become part of query parameter string
+    param_group = parser.add_argument_group('query parameters',
+                                            'parameters for webdata request')
+    param_group.add_argument('--collection',
+                             action=SetQueryParametersAction,
+                             nargs='+',
+                             help='collection identifier')
+    param_group.add_argument('--filename',
+                             action=SetQueryParametersAction,
+                             help='exact webdata filename to download')
+    param_group.add_argument('--crawl',
+                             action=SetQueryParametersAction,
+                             help='crawl job identifier')
+    param_group.add_argument('--crawl-time-after',
+                             action=SetQueryParametersAction,
+                             help='request files with date of creation '
+                                  'during a crawl job after this date')
+    param_group.add_argument('--crawl-time-before',
+                             action=SetQueryParametersAction,
+                             help='request files with date of creation '
+                                  'during a crawl job before this date')
+    param_group.add_argument('--crawl-start-after',
+                             action=SetQueryParametersAction,
+                             help='request files from crawl jobs starting '
+                                  'after this date')
+    param_group.add_argument('--crawl-start-before',
+                             action=SetQueryParametersAction,
+                             help='request files from crawl jobs starting '
+                                  'before this date')
+    return parser
+
+
+def main():
+    parser = _build_parser()
+    args = parser.parse_args()
+
+    if not os.access(args.destination, os.W_OK):
+        msg = 'Cannot write to destination: {}'.format(args.destination)
+        sys.exit(msg)
+
+    # Start log writing process.
+    log_q = multiprocessing.Queue()
+    try:
+        listener = do_listener_logging(log_q, args.log)
+    except OSError as err:
+        print('Could not open file for logging:', err)
+        sys.exit(1)
+
+    # Configure a logger for the main process.
+    try:
+        log_level = [logging.ERROR, logging.INFO, logging.DEBUG][args.verbose]
+    except IndexError:
+        log_level = logging.DEBUG
+    configure_worker_logging(log_q, log_level, 'main')
+
+    # Generate query string for the webdata request.
+    try:
+        query = '?{}'.format(urlencode(args.query_params, safe=':', doseq=True))
+    except AttributeError:
+        query = ''
```

Comment:
That exception is thrown if the user doesn't put in any of the WASAPI query parameters at the commandline, in which case we would want to do the request with no query parameters. I will add a comment.